consensus/ibft/consensus_backend.go

```diff
@@ -225,6 +212,37 @@ func (i *backendIBFT) buildBlock(parent *types.Header) (*types.Block, error) {
 	return block, nil
 }
 
+// calcHeaderTimestamp calculates the new block timestamp, based
+// on the block time and parent timestamp
+func (i *backendIBFT) calcHeaderTimestamp(parentUnix uint64, currentTime time.Time) time.Time {
+	var (
+		parentTimestamp    = time.Unix(int64(parentUnix), 0)
+		potentialTimestamp = parentTimestamp.Add(i.blockTime)
+	)
+
+	if potentialTimestamp.Before(currentTime) {
+		// The deadline for creating this next block
+		// has passed, round it to the nearest
+		// multiple of block time
+		// t........t+blockT...x (t+blockT.x; now).....t+blockT (potential)
+		potentialTimestamp = roundUpTime(currentTime, i.blockTime)
+	}
+
+	return potentialTimestamp
+}
+
+// roundUpTime rounds up the specified time to the
+// nearest higher multiple
+func roundUpTime(t time.Time, roundOn time.Duration) time.Time {
+	roundedTime := t.Round(roundOn)
+
+	if t.Sub(roundedTime) >= 0 {
+		roundedTime = roundedTime.Add(roundOn)
+	}
+
+	return roundedTime
+}
```

Comment:
FYI, you can write like this

```go
func roundUpTime(t time.Time, roundOn time.Duration) time.Time {
	return t.Add(roundOn / 2).Round(roundOn)
}
```

Reply:
Good catch 💯 

Simplified:
[78d0254](https://github.com/0xPolygon/polygon-edge/pull/925/commits/78d02540b764b22a8acb29ed631f2c6097e9b54a)